code/modules/vending/vending.dm

```diff
@@ -617,6 +618,13 @@ ADMIN_INTERACT_PROCS(/obj/machinery/vending, proc/throw_item)
 							account["current_money"] -= product.product_cost
 						else
 							src.credit -= product.product_cost
+					if (!player_list)
```

Comment:
This should all probably be in the above `if(src.pay)` block

Reply:
Above? Wouldn't it make more sense inside since it would only pay out if the money is actually transferred?

Reply:
Yep, that's what I meant